src/query/settings/src/settings_default.rs

```diff
@@ -711,56 +677,55 @@ impl DefaultSettings {
         Ok(Self::instance()?.settings.contains_key(key))
     }
 
-    pub fn convert_value(k: String, mut v: String) -> Result<(String, Option<UserSettingValue>)> {
+    /// Converts and validates a setting value based on its key.
+    ///
+    /// # Arguments
+    /// * `k` - The key of the setting.
+    /// * `v` - The value of the setting to be validated and converted.
+    ///
+    /// # Returns
+    /// * A result containing a tuple of the key and an optional `UserSettingValue` if valid, or an error.
+    pub fn convert_value(k: String, v: String) -> Result<(String, Option<UserSettingValue>)> {
+        // Retrieve the default settings instance
         let default_settings = DefaultSettings::instance()?;
 
+        // Match the setting key with the settings definition
         match default_settings.settings.get(&k) {
+            // Return None for keys that are not found in the settings
             None => Ok((k, None)),
+
+            // Process the setting value for found keys
             Some(setting_value) => {
-                if let Some(possible_values) = &setting_value.possible_values {
-                    let mut checked_value = false;
+                match &setting_value.range {
+                    None => Ok((k, None)),
+                    Some(range) => {
+                        match range {
+                            // Numeric range.
+                            SettingRange::Numeric(_) => {
+                                // Attempt to parse the value as an unsigned 64-bit integer
+                                let u64_val = v.parse::<u64>().map_err(|_| {
+                                    ErrorCode::BadArguments(format!(
+                                        "{} is not a valid u64 value",
+                                        v
+                                    ))
+                                })?;
+                                range.is_within_numeric_range(u64_val)?;
+
+                                // Return the key and value if valid
+                                Ok((k, Some(UserSettingValue::UInt64(u64_val))))
+                            }
+                            // String range.
+                            SettingRange::String(_) => {
+                                // Convert the value to lowercase for case-insensitive comparison
+                                let val_lower = v.to_lowercase();
 
-                    for possible_value in possible_values {
-                        if possible_value.eq_ignore_ascii_case(&v) {
-                            checked_value = true;
-                            v = possible_value.to_string();
-                        }
-                    }
+                                range.is_within_string_range(&val_lower)?;
```

Comment:
```suggestion
                                let matched_value = range.is_within_string_range(&val_lower)?;
                                Ok(k, Some(UserSettingValue::String(matched_value))))
```

Reply:
The suggestion is same as the patch code:
```
                                let val_lower = v.to_lowercase();

                                range.is_within_string_range(&val_lower)?;

                                // Return the key and value if valid
                                Ok((k, Some(UserSettingValue::String(v))))
```

when `is_within_string_range` passed, the `v` is the matched value.